packages/app-studio-toolkit/src/basctlServer/basctlServer.ts

```diff
@@ -5,8 +5,10 @@ import * as _ from "lodash";
 import { _performAction } from "../actions/performer";
 import { ActionsFactory } from "../actions/actionsFactory";
 import { getLogger } from "../logger/logger";
+import { v4 as uuidv4 } from "uuid";
 
-const SOCKETFILE = "/extbin/basctlSocket";
+// generate a unique socket file name on each run
+const SOCKETFILE = `/extbin/basctl-${uuidv4()}.sock`;
```

Comment:
So there is one unique socketfile per each app toolkit activation, right? So for multiple windows in each one there will be a different socketfile

Reply:
exactly. unique socket connection for each extension host